platform/viewer/src/connectedComponents/ToolbarRow.js

```diff
@@ -308,9 +309,17 @@ function _handleBuiltIn(button) {
   }
 
   if (options.behavior === 'DOWNLOAD_SCREEN_SHOT') {
+    const activeViewportIndex = commandsManager.runCommand(
+      'getActiveViewportIndex'
+    );
+
     modal.show({
-      content: ConnectedViewportDownloadForm,
+      content: CornerstoneViewportDownloadForm,
       title: t('Download High Quality Image'),
+      contentProps: {
+        activeViewportIndex,
+        onClose: modal.hide,
+      },
```

Comment:
If we've moved the `CornerstoneViewportDownloadForm` to the `cornerstone` extension, can't we remove this "behavior" entirely? We should be able to replace the toolbar button's command with one that calls `modal.show` instead of using a "built in behavior"

I want us to slowly remove built in behaviors, because they become part of the contract that every "viewer" would need to provide. Offering the more generic, and optional, UI Services seem to be the more versatile pattern.

Reply:
Totally get it. Updated! Thank you.